src/platform/datapath_winuser.c

```diff
@@ -2672,7 +3081,32 @@ CxPlatSocketContextUninitializeComplete(
     SocketProc->Freed = TRUE;
 #endif
 
-    if (SocketProc->Parent->Type == CXPLAT_SOCKET_TCP_LISTENER) {
+    if (SocketProc->Parent->Type != CXPLAT_SOCKET_TCP_LISTENER) {
+        CXPLAT_DBG_ASSERT(SocketProc->RioNotifyArmed == FALSE);
+
+        while (!CxPlatListIsEmpty(&SocketProc->RioSendOverflow)) {
+            CXPLAT_LIST_ENTRY* Entry = CxPlatListRemoveHead(&SocketProc->RioSendOverflow);
+            CXPLAT_SEND_DATA* SendData =
+                CONTAINING_RECORD(Entry, CXPLAT_SEND_DATA, RioOverflowEntry);
+            CxPlatSendDataComplete(SocketProc, SendData, WSA_OPERATION_ABORTED);
+        }
+
+        if (SocketProc->IoStarted) {
+            CXPLAT_DBG_ASSERT(!CxPlatRundownAcquire(&SocketProc->UpcallRundown));
+            CxPlatDataPathRioWorker(SocketProc);
```

Comment:
IMO, it's weird for general socket uninitialize complete to unconditionally call a RIO function, even if it's not using RIO. Should the function be renamed or should we have a RIO check?